src/lightmap.cpp

```diff
@@ -331,6 +331,15 @@ void map::generate_lightmap()
                 }
             }
         }
+        // atomic light is always on
+        std::vector<int> light_indices = v->all_parts_with_feature(VPFLAG_ATOMIC_LIGHT);
```

Comment:
Perhaps add another flag (if it doesn't exist) like `dome_lights_on` and `aisle_lights_on`? This section of the code is on the hot path.